### ISSUE TITLE
Mariko DRAM overvolting

### ISSUE THREAD

**Issue by KazushiMe** [KazushiMe/Switch-OC-Suite#5]
For simplicity of testing, prebuilt loader.kip for AMS 1.2.4 (RAM @ 1894.4/1932.8/1996.8/2064.0/2099.2/2131.2 MHz) and Hekate 5.6.5 (voltage from 605mV to 650mV) are provided here.

[loader.zip](https://github.com/KazushiMe/Switch-OC-Suite/files/7574142/loader.zip)
[hekate.zip](https://github.com/KazushiMe/Switch-OC-Suite/files/7574140/hekate.zip)

Use InfoNX overlay to check if higher voltage is applied.
If you see some improvement in stability at higher frequency, please report here with DRAM chip info.

**Comment by Cooler3D** [comment]
_Hello, and thanks for great work from RU community, you are legend!_

Can I test Hekate with greater voltages, than 0.65V, on my own risk? On my sample of console (Mariko/Samsung) 1996.8 MHz works well even with stock voltages.

**Comment by ms2048** [comment]
Seems like Hynix ones are a lost cause. Mine report as H9HCNNNBKMMLHR-NME 4GB (DRAM-ID in Hekate 5.6.5). Maximum stable freq (tested for many months) is 1862.4. Starting from 1932.8, system either hangs or just crashes during boot (or a few seconds afterwards).

Nothing really changes after overvolting to 650 mV, still the maximum stable freq is 1862.4.

**Comment by ms2048** [comment]
... note however, that in the docs 1795.2 MHz freq is recommended for Hynix. Mine are a little bit better, being stable at 1862.4.

**Comment by Cooler3D** [comment]
SAMSUNG K4U6E3S4AM-MGCJ
![1](https://user-images.githubusercontent.com/13190107/142971050-5a86365d-9ff4-4072-9f66-e7af1ee5a8c1.jpg)
![2](https://user-images.githubusercontent.com/13190107/142971052-126f8d30-8791-4185-a2f1-ee04ed25257c.jpg)
![3](https://user-images.githubusercontent.com/13190107/142971053-6dc20a33-8f3a-47f7-87f6-255b9c8b3aec.jpg)

**Comment by Cooler3D** [comment]
upd. In the process of further testing the overclocking and voltage combinations, Emunand died. It's okay, I have an up-to-date backup, but it might make sense to write a recommendation for everyone in the instructions - to making backup before.

**Comment by 3DSBricker** [comment]
> upd. In the process of further testing the overclocking and voltage combinations, Emunand died. It's okay, I have an up-to-date backup, but it might make sense to write a recommendation for everyone in the instructions - to making backup before.

Perhaps, but he has clear warnings. Your RAM is rated for 1862MHz, mine is too. It works great up to 1996.8MHz without overvolt and 2099MHz with a little overvolt. Didn't test without overvolt though, might test it this week. I couldn't get 2133MHz stable no matter what. It's just the border, you're already overclocking it. RAM overclocking can corrupt things so if you care about your data, make a backup or stop and don't do it.

**Comment by KazushiMe** [comment]
Thanks for all of your feedback. 

So increasing Vddq (MEM I/O) does not help much with current setup. The situation for Erista is similar, where ramping up voltage to 1175mV or higher won't make any difference (default @ 1125mV). 

I wonder if memory timings have something to do with this.

DRAM training (mtc) for mariko is not available in hekate. 

Atmosphere fusee will take over the task for Mariko, or if mtc module is absent for Erista.

But it trains DRAM at fixed 1600 MHz, before pcv module can be loaded. So timings are not correct for OC clock.

Will see if I can patch fusee and upload the diffs & binaries later.

**Comment by KazushiMe** [comment]
DRAM training for OC clockrates:
diff: [mtc-patch.diff.zip](https://github.com/KazushiMe/Switch-OC-Suite/files/7587351/mtc-patch.diff.zip)
binaries: [package3.zip](https://github.com/KazushiMe/Switch-OC-Suite/files/7587363/package3.zip)

**Comment by Cooler3D** [comment]
CAUTION
First blood... I got visual glitches and OS crashes on early achieved freqs. Even with 1996.8 MHz. I hope that is software issue. Trying fix it with backups. Mariko/Samsung.

**Comment by ms2048** [comment]
@Cooler3D before using backups, just turn off your switch and wait 30 minutes. I had the same problem (after trying to overclock toomuch) and it went away by itself.

**Comment by Cooler3D** [comment]
> @Cooler3D before using backups, just turn off your switch and wait 30 minutes. I had the same problem (after trying to overclock too much) and it went away by itself.

**Confirmed**, it works. Thank you!

**Comment by 3DSBricker** [comment]
Is want a typo of won't? Weird. What RAM do you have? Check Hekate, I don't mean the manufacturer (Samsung) but the long number. It should end with either AA or AM and then a dash (-) and some other characters. Just say whether it's AA or AM.

**Comment by Cooler3D** [comment]
_> SAMSUNG K4U6E3S4AM-MGCJ_
Typo. Fixed.
Sorry, my English is not too good.

**Comment by 3DSBricker** [comment]
Doesn't matter. AM means it's rated for 1862MHz (or 1866MHz I'm not sure about LPDDR versions), I have the same. Anything above it is an overclock. Meaning may the best chip win. I only now noticed you already mentioned the RAM model, by the way. I don't know why previously working frequencies stopped working. Are you sure it's 1996.8 with no over-, or undervolt?

**Comment by Cooler3D** [comment]
> binaries: [package3.zip](https://github.com/KazushiMe/Switch-OC-Suite/files/7587363/package3.zip)

I do not know how this is related, but with this binary I again managed to successfully boot with the above memory overclocking.

**Comment by 3DSBricker** [comment]
It is related, it is some modification with RAM timings. I don't know why you previously could do it with the stock timings, but at least it works again.

**Comment by Cooler3D** [comment]
Benchmarked again, with new package3 binary. Looks like latency numbers the same, as before.

<details>

![2021112320211400-0004EDF51B27297FAD35745CCDCC73BD](https://user-images.githubusercontent.com/13190107/143073823-d1e49c75-7293-423b-b060-f8dad867fdec.jpg)

</details>

<details>

![2021112320164200-0004EDF51B27297FAD35745CCDCC73BD](https://user-images.githubusercontent.com/13190107/143073819-2c15bbfa-c146-40a1-88d4-0e3644e5fdc3.jpg)

</details>

**Comment by 3DSBricker** [comment]
Looks more like it's a little faster. I assume it's not stable if it crashed. Is that the 2099@650mV again?

**Comment by Cooler3D** [comment]
> Are you sure it's 1996.8 with no over-, or undervolt?

Yes. Works absolutely stable from September, tested on lot of heavy games with overclock and graphic mods. For example, tens of hours in Wolfenstein II: The New Colossus, modded to native resolution and 60FPS, in portable (without attached charger), at freqs CPU ~1400/ GPU ~1300g/ RAM ~2000.

**Comment by Cooler3D** [comment]
605 mv / 2064 MHz works stable :D

**Comment by 3DSBricker** [comment]
> 2064 MHz works stable at 605 mv... But 2098 MHz unstable even at 650 mv. Really, silicon lottery? :D
> 
> Still worth it!

You're still running 200MHz above its rated frequency with only 5mV extra. (are you sure it's fully stable?) and even another 264MHz above Nintendo's maximum. A total of 464MHz, or 928MT/s extra!

**Comment by Cooler3D** [comment]
> > 2064 MHz works stable at 605 mv... But 2098 MHz unstable even at 650 mv. Really, silicon lottery? :D
> > Still worth it!
> 
> You're still running 200MHz above its rated frequency with only 5mV extra. (are you sure it's fully stable?) and even another 264MHz above Nintendo's maximum. A total of 464MHz, or 928MT/s extra!

Yes, i am sure:

![2021112321440900-0004EDF51B27297FAD35745CCDCC73BD](https://user-images.githubusercontent.com/13190107/143231777-a430da7a-d244-48d6-85b5-1e3610a25146.jpg)

![2021112321422100-0004EDF51B27297FAD35745CCDCC73BD](https://user-images.githubusercontent.com/13190107/143231782-edab2d19-d6c5-4c0b-a369-d4179300c8dd.jpg)

**Comment by 3DSBricker** [comment]
Well you can never be fully sure. That's why you always need to be ready for the moment you start your Switch, and it doesn't boot correctly anymore, crashes or is just fully corrupted. Always make backups. It has also been in the source code since the moment this all started. I understand not everyone looks in the comments of the source code, but there were warnings... (neither is anyone responsible for anything that might happen). Anything you do has the risk of damaging things.

**Comment by Cooler3D** [comment]
> Well you can never be fully sure. That's why you always need to be ready for the moment you start your Switch, and it doesn't boot correctly anymore, crashes or is just fully corrupted. Always make backups. It has also been in the source code since the moment this all started. I understand not everyone looks in the comments of the source code, but there were warnings... (neither is anyone responsible for anything that might happen). Anything you do has the risk of damaging things.

Thanks, don't worry, I always try to keep my backups fresh. Besides, like the others, I'm not here by chance. I am well aware of the danger of testing the alpha version of the program to overclock the console.

If something bad happens during testing, I will definitely let you know. I confirm stability not earlier than after several hours of playing high-load projects.

**Comment by 3DSBricker** [comment]
> > Well you can never be fully sure. That's why you always need to be ready for the moment you start your Switch, and it doesn't boot correctly anymore, crashes or is just fully corrupted. Always make backups. It has also been in the source code since the moment this all started. I understand not everyone looks in the comments of the source code, but there were warnings... (neither is anyone responsible for anything that might happen). Anything you do has the risk of damaging things.
> 
> Thanks, don't worry, I always try to keep my backups fresh. Besides, like the others, I'm not here by chance. I am well aware of the danger of testing the alpha version of the program to overclock the console.
> 
> If something bad happens during testing, I will definitely let know here. I confirm stability not earlier than after several hours of playing high-load projects.

That's good. I saw some people in GBATemp clearly having zero knowledge about what they were doing. I'd say just don't do this if you don't know what you're doing. Feedback is always helpful for others, and yourself too.

**Comment by ms2048** [comment]
> _Nov 24_
> LPDDR4X Vddq I/O voltage is proved to set correctly and atmosphere mtc at boot does not seem to have any effect. The last resort is tinkering timing tables.

Should I retest then with modified package3 or would such test be worthless?

**Comment by Cooler3D** [comment]
Ok, finally, i got rock stable **2099 MHz at memory with 650 mv.**
With **605 mv. my Switch stable at 2064 MHz.**
**Without any overvolting - my handheld works great with 1996 MHz** on memory.
Bootloop with 2131 MHz.
 
I would also like to try my luck with higher voltages than 650mV. For example 700 mv - 750 mv. I do not see any danger for the memory itself, except perhaps for the hub and system logic. Anyway, I modified the console cooling system, and I believe that nothing terrible should happen.

**Comment by KazushiMe** [comment]
> Should I retest then with modified package3 or would such test be worthless?

No, that's for memory training at pkg1 state, which doesn't help much.

<hr>

To tinker mtc timing tables in pcv module, stub memory training at boot in `fusee/program/source/mtc/fusee_mtc_mariko.cpp`, replace ldr_patcher.cpp with [patch_mtc_loader.zip](https://github.com/KazushiMe/Switch-OC-Suite/files/7600567/patch_mtc_loader.zip), and rebuild package3.

```cpp
/* Patch parameters, examples: */
mtc_table->min_volt = 887;
mtc_table->burst_regs_arr[7] = 42; //emc_r2w  
```

As DRAM is initialized in pkg1, for patching sdram parameters, edit `fusee/program/sdram_params/bin/sdram_params_mariko_[0-12].bin` manually with hex editor, then run `fusee/program/update_sdram_params.py` and rebuild fusee.

**Comment by ms2048** [comment]
I probably won't be of much help then, as unfortunately I don't have any env for atmosphere compilation and my free time is limited.

On the other hand, whenever it is simply the case of swapping some files, I am always willing to risk the well being of my switch, no problem there :)

**Comment by Cooler3D** [comment]
> I probably won't be of much help then, as unfortunately I don't have any env for atmosphere compilation and my free time is limited.
> 
> On the other hand, whenever it is simply the case of swapping some files, I am always willing to risk the well being of my switch, no problem there :)

I will try to rebuild soon. Going to post compilated result here.

**Comment by 3DSBricker** [comment]
> Ok, finally, i got rock stable **2099 MHz** at memory with **650 mv.** With **605 mv.** my Switch stable at **2064 MHz.** **Without any overvolting** - my handheld works great with **1996 MHz** on memory. Bootloop with 2131 MHz.
> 
> I would also like to try my luck with higher voltages than 650mV. For example 700 mv - 750 mv. I do not see any danger for the memory itself, except perhaps for the hub and system logic. Anyway, I modified the console cooling system, and I believe that nothing terrible should happen.

Sure. But you know what I said what happened to mine. Just try 660mV first. If that does weird even with previously stable frequency you definitely shouldn't go higher.

**Comment by Cooler3D** [comment]
> 

Hm... I remember, that we have the same memory. Don't have any precompiled binaries left? Do you still have tested binaries, with higher voltage? Can i try them?

**Comment by 3DSBricker** [comment]
It's very easy to make. Just apply the diff file and change 600 to 660. (ignore the other zero's behind it, just leave them). I won't share them as it's above the (still somewhat) safe Voltage.

**Comment by Cooler3D** [comment]
After doing a lot of new tests, I came to the following conclusion:
700 mv - works without any issues even in hardcore scenarios, and without noticeable power consumption increasing.

**Comment by 3DSBricker** [comment]
Great to hear. I wonder why mine acted weird. I finally applied thermal paste into the RAM chips so I might try something like 670mV.

**Comment by 3DSBricker** [comment]
Does 2133 work with those (insane) Voltages?

**Comment by Cooler3D** [comment]
> Does 2133 work with those (insane) Voltages? Update on mine: even with 750mV 2133 doesn't work. It looks stable for some time but still crashes.

I didn't have time to figure out the compilation of memory timings, so I haven't tried 2133 yet. I used 750 millivolts at a frequency of 2099 to test stability. Your statement on this matter raised my doubts, since I do not observe any grounds for unstable operation at such low voltages.

By the way, the stable operation of the memory at a frequency higher than 2133 megahertz is clearly not hampered by the lack of voltage. 750 millivolts, as far as i know, should be more than enough.

**Comment by Cooler3D** [comment]
@KazushiMe 
Hello! Thanks again for your invaluable contributions to the community.
I do not presume to say this, but looking at the situation through the prism of my overclocking experience, I have formed a strong feeling that the preset for 2131 is broken, as if something is wrong with it. I have rechecked the documentation for the JEDEC standard, and I ask you to forgive me for a possibly stupid question, but could there have been an error in calculating the coefficient? Is 2131 exactly? I have never met a memory at a similar frequency. How about 2133?

**Comment by KazushiMe** [comment]
@Cooler3D I don't have access to JEDEC specs as those are behind paywall. All these DRAM clocks are sourced from hekate: https://github.com/CTCaer/hekate/blob/master/modules/hekate_libsys_minerva/sys_sdrammtc.c#L68.

**Comment by 3DSBricker** [comment]
800mV is a very, very bad idea.

**Comment by Cooler3D** [comment]
> @Cooler3D I don't have access to JEDEC specs as those are behind paywall. All these DRAM clocks are sourced from hekate: https://github.com/CTCaer/hekate/blob/master/modules/hekate_libsys_minerva/sys_sdrammtc.c#L68.
Thanks. Looks genuine.
I tried to make sure with the help of a differential equation... There was no need, false version, sorry.

**Comment by Cooler3D** [comment]
The last roll of luck. 850 mv. Almost stable at 800 mv, but not enough.

**Comment by Cooler3D** [comment]
Looks like TDP did not changed at all. The same thermals reports, 33C/32C/34C (with stock CPU\GPU frequencies in Slay the Spire)

**Comment by 3DSBricker** [comment]
I'm really concerned about the lifespan of your RAM. It's made for up to 650mV. 750mv would be comparable to the DDR4 XMP Voltages. Over the stock 1.2V to 1.35V. Going higher can be dangerous. You didn't just went another 50mV up, you went 100mV up. That's like 1.45V for DDR4. Your RAM won't last long with those kind of Voltages, I'd recommend you to lower it. That 33MHz really isn't worth so much extra Voltage.

**Comment by 3DSBricker** [comment]
> Looks like TDP did not changed at all. The same thermals reports, 33C/32C/34C (with stock CPU\GPU frequencies in Slay the Spire). I still did not corrected timings. But for now, it just works. More demanding games will be checked coming soon. UPD. Crashed. Well, enough for my luckiness-capacity for today. Any way, even if it can work on 850 mv, that is too much for everyday scenarios. 2099 works stable as a... stool, with 620 mv.

It crashed with 42% extra Voltage? I think 2133 just isn't doable on these chips. Even chips made for it can't reach it.

**Comment by Cooler3D** [comment]
> I'm really concerned about the lifespan of your RAM. It's made for up to 650mV. 750mv would be comparable to the DDR4 XMP Voltages. Over the stock 1.2V to 1.35V. Going higher can be dangerous. You didn't just went another 50mV up, you went 100mV up. That's like 1.45V for DDR4. Your RAM won't last long with those kind of Voltages, I'd recommend you to lower it. That 33MHz really isn't worth so much extra Voltage.

As far as i know, not voltage, but heat can bring damage to silicon components. Of course, increasing voltage equaly to increasing heat generation. But i do not think, that memory can by fried without increasing any other temperatures, like SOC etc. 33C/32C/34C, that is idle.

**Comment by Cooler3D** [comment]
> > Looks like TDP did not changed at all. The same thermals reports, 33C/32C/34C (with stock CPU\GPU frequencies in Slay the Spire). I still did not corrected timings. But for now, it just works. More demanding games will be checked coming soon. UPD. Crashed. Well, enough for my luckiness-capacity for today. Any way, even if it can work on 850 mv, that is too much for everyday scenarios. 2099 works stable as a... stool, with 620 mv.
> 
> It crashed with 42% extra Voltage? I think 2133 just isn't doable on these chips. Even chips made for it can't reach it.

Maybe main issue, in my case, in default timings - which generated for 1600 mHz value?

**Comment by 3DSBricker** [comment]
I understand what you're saying so don't worry about that. Do worry about your RAM though. Take an example to CPU's:
https://youtu.be/g1Yp6pQRdns
Start watching at 3:52. Yes you will likely hit thermal limits before Voltage, but you don't know what temperature your RAM is. Neither will Voltage or heat actually kill or degrade it. It's just something that accelarates it. Mostly for CPU's it's the Amperage that kills or fastly degrades it. More Voltage is more headroom for Amperage. More heat is faster degradation. It's always like that. Not only with CPU's (even when electrocuting living things Amperage is the thing that kills).

**Comment by Cooler3D** [comment]
> I understand what you're saying so don't worry about that. Do worry about your RAM though. Take an example to CPU's: https://youtu.be/g1Yp6pQRdns Start watching at 3:52. Yes you will likely hit thermal limits before Voltage, but you don't know what temperature your RAM is. Neither will Voltage or heat actually kill or degrade it. It's just something that accelarates it. Mostly for CPU's it's the Amperage that kills or fastly degrades it. More Voltage is more headroom for Amperage. More heat is faster degradation. It's always like that. Not only with CPU's (even when electrocuting living things Amperage is the thing that kills).

Thanks, but as i say before, do not worry much. I have "some" experience and knowledge base in cooling/overclocking, thermal\simulations etc. Sorta HW "enthusiast":
For example, years before (in 2015), I became the first in the world to own a system based on MiniITX with size of an regular Xbox One, based on an 18-core Intel Zeon (2699v3), overclocked by FSP, with unlocked turbo boost in all cores load, paired with an overclocked 980TI (later 1080TI). And that system did not reach throttling even with a parallel load of all components/cores by 100%.
Two years ago, I became the first in the world to own a system based on regular ATX case, with full integrated (inside case) cooling, which can effective dispel more than 2000 watt TDP 24/7. (AMD 3990x manually (per TTX) overclocked to ~4.3GHz-4.5GHZ in 100% load to all cores scenarios, which means almost 1000 watt only from CPU socket).
Of course, much of custom work done to cooling for achieve such results for regular exploitations 24/7, and that is only 2 example from my experience. But I think you understand me, and that I know, what I doing, when setting 850mv on Switch EMC regulator.

**Comment by 3DSBricker** [comment]
Yes I understand. That still doesn't make it less risky. It's also about whether it's worth it. If 2099 works within the normal Voltages and you need over 42% extra Voltage.... It's always your choice but I just don't think it's worth it. Cooling a 1000W, though. Very impressive.

**Comment by Cooler3D** [comment]
> Yes I understand. That still doesn't make it less risky. It's also about whether it's worth it. If 2099 works within the normal Voltages and you need over 42% extra Voltage.... It's always your choice but I just don't think it's worth it. Cooling a 1000W, though. Very impressive.

Of course, I did this for testing purposes only. Just to clarify additional heat amount. With lower speeds, memory can work with that voltage without temperature/heat issues. Good to know for feature tests with edited timings. Such a negligible increase in speed does not at all justify the required overvoltage. For permanent using i preferred 2064MHz. I was surprised to find that this frequency works without any increase in voltage, 600mv.

**Comment by KazushiMe** [comment]
Preliminary mtc timing patching is implemented.
https://github.com/KazushiMe/Switch-OC-Suite/blob/0823cf54bc237658ac2cceb37bb969e932bc3ad0/Source/Atmosphere/stratosphere/loader/source/ldr_patcher.cpp#L187-L287

Timing parameters that somewhat make sense will be adjusted proportionally.
Clock divisor DIVM and DIVN is re-calculated as well.

Ideally, all dram chips should get 2131.2 MHz working with moderate Vddq overvolting. So only 2131.2 MHz loader.kip is provided: [1.2.5_loader_2131.2_adjusted.kip.zip](https://github.com/KazushiMe/Switch-OC-Suite/files/7644465/1.2.5_loader_2131.2_adjusted.kip.zip)

**Comment by renA21** [comment]
Tested the build above, my unit refuses to boot and the Switch logo doesn't show up at all even when I have set voltage to 650mV.

**Comment by Cooler3D** [comment]
> Tested the build above, my unit refuses to boot where the Switch logo doesn't show up at all even when I have set voltage to 650mV.

The same behavior in my case.

**Comment by Cooler3D** [comment]
I checked last update. GPU overclocking works as it should, but loader.kip files now did not to anithing. 1366MHz on memory permanently.

**Comment by KazushiMe** [comment]
Sys-clk is updated with RAM frequency selection, so that ths system won't be stuck at boot if DRAM clock or timings are not set properly.

**Comment by Cooler3D** [comment]
> 

Yes, i got it, but changing to MAX RAM frequency from manager does not changes actual frequency. I checked every kip loader, every time only 1333MHz at ram.

**Comment by MajinChibi** [comment]
The same, it just stays at 1333 mhz

**Comment by KazushiMe** [comment]
Finally got my device back. 
Here are some superficial results on Hynix DRAM with timings and clock divisors adjusted:

|  | Timings adjusted (partially) | Clock divisors adjusted | Benchmark |
| :-: | :-: | :-: | :-: |
| 1 | ❌ | ❌ | ![2021111302230800-F1C11A22FAEE3B82F21B330E1B786A39](https://user-images.githubusercontent.com/85604869/145620604-d822eae2-b120-465a-906b-ca41707d6dc5.jpg) |
| 2 | ❌ | ✅ | ![2021121101474400-58651EB6190D84545B23A9E085BB315E](https://user-images.githubusercontent.com/85604869/145620795-5bb609fb-3274-4005-b49b-45b5f8b310c3.jpg) |
| 3 | ✅ | ✅ | ![2021121101573500-58651EB6190D84545B23A9E085BB315E](https://user-images.githubusercontent.com/85604869/145620857-2021a69d-b943-4f47-a554-2b84d3d89874.jpg) |

```cpp
#ifdef ADJUST_TIMINGS
/* Partial, may not include all the timings. */
ADJUST_PROP_WITHIN_ALL_REG(mtc_table_1600, mtc_table_1331, emc_rc);
ADJUST_PROP_WITHIN_ALL_REG(mtc_table_1600, mtc_table_1331, emc_rfc);
ADJUST_PROP_WITHIN_ALL_REG(mtc_table_1600, mtc_table_1331, emc_rfcpb);
ADJUST_PROP_WITHIN_ALL_REG(mtc_table_1600, mtc_table_1331, emc_ras);
ADJUST_PROP_WITHIN_ALL_REG(mtc_table_1600, mtc_table_1331, emc_rp);
ADJUST_PROP_WITHIN_ALL_REG(mtc_table_1600, mtc_table_1331, emc_w2r);
ADJUST_PROP_WITHIN_ALL_REG(mtc_table_1600, mtc_table_1331, emc_r2p);
ADJUST_PROP_WITHIN_ALL_REG(mtc_table_1600, mtc_table_1331, emc_w2p);
ADJUST_PROP_WITHIN_ALL_REG(mtc_table_1600, mtc_table_1331, emc_trtm);
ADJUST_PROP_WITHIN_ALL_REG(mtc_table_1600, mtc_table_1331, emc_twtm);
ADJUST_PROP_WITHIN_ALL_REG(mtc_table_1600, mtc_table_1331, emc_tratm);
ADJUST_PROP_WITHIN_ALL_REG(mtc_table_1600, mtc_table_1331, emc_twatm);

ADJUST_PROPORTIONAL(mtc_table_1600, mtc_table_1331, min_mrs_wait);
ADJUST_PROPORTIONAL(mtc_table_1600, mtc_table_1331, latency);
#endif

#ifdef ADJUST_DIVS
mtc_table_1600->pllmb_divm = divm;
mtc_table_1600->pllmb_divn = divn;
#endif
```

I don't have time to investigate and test thoroughly.

Modify "ldr_patcher.cpp" and build loader.kip if you'd like to try at higher frequencies / with other timing combinations, then see if it could alleviate screen artifacts or instability.

**Comment by MajinChibi** [comment]
Wow thank you very much!
Mariko with hynix:

my self compiled loader, just changed the frequency to 1862 Mhz
![before](https://user-images.githubusercontent.com/17251655/145854311-811fd9b5-f065-4bea-bbd3-0d6e564d9beb.jpg)
![before_lat](https://user-images.githubusercontent.com/17251655/145854341-3a6f5acc-eb23-40ae-b62f-c5f491b8f8c9.jpg)

your loader with adjusted timings:
![after](https://user-images.githubusercontent.com/17251655/145854560-18996ae8-244e-4bff-9630-434d3a747b19.jpg)
![after_lat](https://user-images.githubusercontent.com/17251655/145854582-75045b13-d393-4dd0-a16f-248c1d07f8e5.jpg)

Now i give it a few loops with the mem tester.

**Comment by 3DSBricker** [comment]
Do remember, as said in the Readme, that MemTesterNX is not the best way to test stability. Games like BOTW are better. I tested mine and it passed many loops. Tested BOTW and it froze and nearly got corrupted. Hence the need of backups.

**Comment by Cooler3D** [comment]
![2021121610222100-0004EDF51B27297FAD35745CCDCC73BD](https://user-images.githubusercontent.com/13190107/146359675-80fdcc74-bd5a-4c38-a0c8-b343895d5757.jpg)
![2021121610240100-0004EDF51B27297FAD35745CCDCC73BD](https://user-images.githubusercontent.com/13190107/146359677-5ef7e714-6365-41c7-a6bc-9092eae4d99a.jpg)
Tested on The Witcher 3, will do more tests on BOTW soon.

**Comment by 3DSBricker** [comment]
Is yours stable at 2131MHz?

**Comment by Cooler3D** [comment]
> Is yours stable at 2131MHz?

Tested few hours on RE V and Witcher 3. Works well without noticeable issues.
Of course, with slight memory overvolting (615).

**Comment by renA21** [comment]
Mine's not stable at 2131MHz at all with max overvoltaing when playing games, just instant freeze when loaded in-game for a few seconds on GTA.

**Comment by 3DSBricker** [comment]
> > Is yours stable at 2131MHz?
> 
> Tested few hours on RE V and Witcher 3. Works well without noticeable issues. Of course, with slight memory overvolting (615).
> 
> P.s. Found that 2064 works great without overvolting, lol.

1996.8MHz is normalized 2000MHz. Still custom, but normalized. I prefer 1996.8MHz as 2099MHz stopped working stably after the latest improment in timings. 1862.4MHz works great too, though. 1862 and 2133 are the only official frequencies.

**Comment by Cooler3D** [comment]
> 2099MHz

2099MHz is not works stable for my (without overclocking), but works fine with little overvoltage.
1996MHz works without overvoltage, 2064MHz too.
2131MHz (old and improved by timings) works only with overvolting. Light in case with adjusted timings, and huge overvolting in case without retiming. Results 5 message above was my best stable combination at moment.
Just for lulz i tied undervolt memory. When i try voltages below 595mv (like 590mv) InfoNX shows 600mv.

**Comment by 3DSBricker** [comment]
> > 2099MHz
> 
> 2099MHz is not works stable for my (without overclocking), but works fine with little overvoltage. 1996MHz works without overvoltage, 2064MHz too. 2131MHz (old and improved by timings) works only with overvolting. Light in case with adjusted timings, and huge overvolting in case without retiming. Results 5 message above was my best stable combination at moment. Just for lulz i tied undervolt memory. When i try voltages below 595mv (like 590mv) InfoNX shows 600mv. About undervolting... It would be nice to be able to adjust the undervolting value as a offset (for CPU and GPU voltages). Almost all processors can work fine with undervolting offset in 0.05-0.15v. Maybe Tegra can too.

I tried. RAM undervolting works for me up to 500mV. Haven't really tested stability, was just for KazushiMe to confirm that Voltage is being set. About CPU/GPU undervolting, ARM isn't like X86. If it has too little Voltage, it falls back to the previous frequency. For CPU there's a homebrew app to test if the frequency is being set as InfoNX/Status Monitor aren't always correct. They mention what's being set, for RAM it's always correct. GPU too, if too little Voltage then it just won't set (even Sys-CLK will say it's the previous value). You could try 2397 for example. Voltage is already dynamic so I don't see why it would be really useful. (2397 idle can be as low as 800mV and under load as high as 1220mV)

**Comment by Cooler3D** [comment]
![2022022114155700-0004EDF51B27297FAD35745CCDCC73BD](https://user-images.githubusercontent.com/13190107/154952910-1e0c45a5-ea4b-465b-8a7e-4b14bc1030e5.jpg)
![2022022114180100-0004EDF51B27297FAD35745CCDCC73BD](https://user-images.githubusercontent.com/13190107/154952916-20e18e14-2350-420a-85f1-bcd55f00f81a.jpg)

**Comment by zal16** [comment]
The Switch OLED contains K4U6E3S4AA-MGCL RAM chip, he is rated for 4266 mbps. It's possible to add a 2133 MHz option for this console?

**Comment by zal16** [comment]
I only need to modify the "marikoEmcMaxClock":1996800 to 2131.2 (or other values divisible for 9600?

**Comment by NaGaa95** [comment]
> I only need to modify the cust_conf section on "marikoEmcMaxClock":1996800 to 2131200 (or other values divisible by 9600)?

Yep, you can also change mtcConf to 1 for better timing on the ram

As always ram overclocking can be dangerous and corrupt your nand so try it first on Emunand, and for stress test i recommand you to test it in a game like BOTW, not MemTester 
Good luck !

**Comment by halop** [comment]
Hi,

i've tested many configs with Memtester & a few hours gameplay (BotW).

RAM: Samsung K4U6E3S4AA-MGCL (Switch OLED)

AUTO_ADJ_MARIKO_SAFE & AUTO_ADJ_MARIKO_4266 with both timings my Switch is only stable at 1932 MHz without OC
With no adjustment (NO_ADJ_ALL) my Switch is stable at 2099 MHz without OC.



With OC (+50mV) the RAM can reach 2131 MHz with AUTO_ADJ_MARIKO_4266 & NO_ADJ_ALL timings, but is not stable enough.
( crashes after a few hours gameplay).



The standard timings seems to be a better fit for this RAM:

2131 MHz | OC 650 mV
DRAM Timing 1
AUTO_ADJ_MARIKO_4266:

![DRAM Timing 1](https://user-images.githubusercontent.com/4215938/215253035-cb54009e-32a2-4a5d-bf21-f47de43fb4b6.jpg)


2131 MHz | no OC
DRAM Timing 0 
NO_ADJ_ALL:

![DRAM Timing 0](https://user-images.githubusercontent.com/4215938/215253045-eb924349-471a-4e47-a53f-2b1884994905.jpg)



2099 MHz | no OC
DRAM Timing 0
NO_ADJ_ALL:

![2099 no OC](https://user-images.githubusercontent.com/4215938/215253360-8305ef1d-a420-47b4-bc6f-63ead443bc12.jpg)

![2099 no OC ms](https://user-images.githubusercontent.com/4215938/215253364-3627fec3-6add-4aa3-b303-0adc4db3adc6.jpg)


thanks for this awesome OC Suite!

**Comment by Cooler3D** [comment]
SAMSUNG K4U6E3S4AM-MGCJ 3733 Rated, 1200mV / 550mV - According to the first day of gaming experience, it is quite stable.

![2](https://user-images.githubusercontent.com/13190107/218334750-1c5f6ee1-7d54-458e-803c-51aa1f330a37.png)
![1](https://user-images.githubusercontent.com/13190107/218334752-4702fc01-f53b-46fe-8041-0f101dd996d9.png)

It seems that the memtester is practically useless at such frequencies, since the CPU capabilities are the bottleneck in it.

At first I even thought that the increase in frequency does not provide an effective performance boost, but gaming tests quickly put everything in its place.
